### PR TITLE
Ankit | Test | Fetch Go cardless integrated bank balance in dashboard page

### PR DESCRIPTION
**Pull request by ankitdubeyoffice**
https://app.clickup.com/t/86cx4gg7h

## Release Notes

- **New Features**
	- Added a button for refreshing bank accounts, conditionally displayed based on user context.
	- Introduced a section for connecting or reconnecting bank accounts with improved UI elements.
	- Enhanced responsiveness for mobile users in the ledger view.

- **Bug Fixes**
	- Improved conditional rendering of bank balance information.

- **Documentation**
	- Updated localization files with new key-value pairs for better language support.

- **Chores**
	- Added a new API endpoint for refreshing bank accounts.

### CODE REVIEW

File: apps/web-giddh/src/app/ledger/ledger.component.html [hunk 1]
```diff
@@ -476,6 +476,20 @@
...
+                                        href="javascript:void(0)"
+                                        id="share"
+                                        class="icomoon-ico d-flex align-items-center"
+                                        [matTooltip]="commonLocaleData?.app_refresh"
+                                        [matTooltipPosition]="'above'"
+                                        matTooltipClass="tooltip-black"
+                                        (click)="refreshBank()"
+                                    >
+                                        <i aria-hidden="true" class="icon-refresh"></i>
+                                    </a>
```
[the remark this answers is not in the document]
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/ledger/ledger.component.html]: All Done , Please check

File: apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.scss [hunk 2]
```diff
@@ -46,4 +46,7 @@ p {
 }
 .width-86 {
     width: 86% !important;
+}
+.refersh-pd{
+    padding: 15px 12px;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.scss:52]: format code

File: apps/web-giddh/src/app/home/home.store.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import { Injectable } from "@angular/core";
+import { ComponentStore, tapResponse } from "@ngrx/component-store";
+import { Observable, switchMap, catchError, EMPTY } from "rxjs";
+import { ToasterService } from "../services/toaster.service";
+import { ContactService } from "../services/contact.service";
+
+export interface HomeState {
+    bankMessage: any;
+    isBankRefreshing: boolean
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/home/home.store.ts]:
```suggestion
    isBankRefreshing: boolean;
```

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 4]
```diff
@@ -352,7 +357,8 @@ export class LedgerComponent implements OnInit, OnDestroy {
         private pageLeaveUtilityService: PageLeaveUtilityService,
         private router: Router,
         private settingsIntegrationService: SettingsIntegrationService,
-        private componentStore: BankIntegrationComponentStore
+        private componentStore: BankIntegrationComponentStore,
+        private homeComponentStore: HomeComponentStore,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/ledger/ledger.component.ts]:
```suggestion
        private homeComponentStore: HomeComponentStore
```

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 5]
```diff
@@ -997,6 +1003,13 @@ export class LedgerComponent implements OnInit, OnDestroy {
                 this.getBankTransactions();
             }
         };
+
+        this.bankMessage$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response === BANK_REFRESH_SUCCESS_MESSAGE) {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/ledger/ledger.component.ts]: change this logic show, we are getting message by API response

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 6]
```diff
@@ -3144,4 +3157,13 @@ export class LedgerComponent implements OnInit, OnDestroy {
         });
     }
 
+    /**
+     * Refresh bank
+     *
+     * @memberof LedgerComponent
+     */
+    public refreshBank() {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/ledger/ledger.component.ts]: Refresh Bank Transactions

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.scss (hunk 2 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.scss:52]: Done

File: apps/web-giddh/src/app/home/home.store.ts (hunk 3 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/home/home.store.ts]: Done

File: apps/web-giddh/src/app/ledger/ledger.component.ts (hunk 4 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/ledger/ledger.component.ts]: Done

File: apps/web-giddh/src/app/ledger/ledger.component.ts (hunk 5 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/ledger/ledger.component.ts]: Done

File: apps/web-giddh/src/app/ledger/ledger.component.ts (hunk 6 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/ledger/ledger.component.ts]: Done

**Review by dilpreetsinghofficial — APPROVED** [review]

**Pull request merged**